hrjobcontract/CRM/Hrjobcontract/BAO/HRJobDetails.php

```diff
@@ -81,6 +81,135 @@ public static function getRecordCount($params) {
   }
 
   /**
+   * Check if given Contract start and end dates are available for given Contact.
+   * 
+   * @param array $params key-value pairs
+   * @return bool
+   */
+  public static function validateDates(array $params) {
+    if (empty($params['contact_id'])) {
+      throw new CiviCRM_API3_Exception("Please specify 'contact_id' value.");
+    }
+    if (empty($params['period_start_date'])) {
+      throw new CiviCRM_API3_Exception("Please specify 'period_start_date' value.");
+    }
+    $contactId = $params['contact_id'];
+    $periodStartDate = date('Y-m-d', strtotime($params['period_start_date']));
+    $periodEndDate = !empty($params['period_end_date']) ? date('Y-m-d', strtotime($params['period_end_date'])) : NULL;
+    $jobContractId = !empty($params['jobcontract_id']) ? $params['jobcontract_id'] : null;
+    $conflictingContracts = self::getConflictingContracts($contactId, $periodStartDate, $periodEndDate, $jobContractId);
+    if (empty($conflictingContracts)) {
+      return array(
+        'success' => TRUE,
+        'message' => NULL,
+      );
+    }
+    return array(
+      'success' => FALSE,
+      'message' => self::getConflictMessage($conflictingContracts),
+    );
+  }
+
+  /**
+   * Return an array of Job Contracts conflicting with given Contact ID,
+   * Period Start Date, Period End Date and optional Job Contract Id.
+   * 
+   * @param int $contactId
+   * @param string $periodStartDate
+   * @param string $periodEndDate
+   * @param int $jobContractId
+   * @return array
+   */
+  public static function getConflictingContracts($contactId, $periodStartDate, $periodEndDate, $jobContractId = null) {
+    $conflictingContracts = array();
+    $conflictingContractsQuery = "
+      SELECT jc.id, jcd.title, jcd.period_start_date, jcd.period_end_date, jcd.jobcontract_revision_id, jcr.effective_date
```

Comment:
this query should exclude deleted contracts and revisions .. the contracts and revisions that has "deleted" column set to 1